appdev/guides/dependencies.rst

```diff
@@ -0,0 +1,136 @@
+
+Put the libraries source code at ``libs/LIBNAME``, because this is where clickable will look for it by default. If the dependency source code is available as a git repository, it is a good idea to add it as a `git submodule <https://git-scm.com/book/de/v1/Git-Tools-Submodule>`_. Otherwise you might want to add a script to download the sources.
+
+Add a `libraries section <http://clickable.bhdouglass.com/en/latest/clickable-json.html#libraries>`_ to your clickable.json, like this:
+
+.. code-block:: JSON
+   :emphasize-lines: 4-8
+
+    {
+      "template": "cmake",
+      "dir": "build/app",
+      "libraries": {
+        "LIBNAME": {
+          "template": "cmake"
+        }
+      }
+    }
+
+If the library does not contain a CMake configuration, you may use the `qmake` or `custom` template instead. If you choose `custom`, you need to add a `build` command.
+
+Note, that you should define a custom app build directory via the ``dir`` param as shown above. Otherwise a `clickable clean` (which is part of the default `clickable` command) would delete the library's build directory, too.
+
+Optionally, configure the compilation by adding ``build_args``, which may look like this:
+
+.. code-block:: JSON
+   :emphasize-lines: 7-12
+
+    {
+      "template": "cmake",
+      "dir": "build/app",
+      "libraries": {
+        "LIBNAME": {
+          "template": "cmake",
+          "build_args": [
+            "-DBUILD_EXAMPLES=OFF",
+            "-DBUILD_DOCS=OFF",
+            "-DBUILD_TESTS=OFF",
+            "-DBUILD_SHARED_LIBS=OFF"
+          ]
+        }
+      }
+    }
+
+Build arguments are usually project specific. Therefore, study the library's build instructions and also look for ``option`` settings in its ``CMakeLists.txt`` (if CMake is used). The default build directory is ``build/LIBNAME/ARCHITECTURE``. Therefore builds for different architectures can exist in parallel.
+
+To actually build the library run ``clickable build-libs``. You may want to build it for the desktop, too, via ``clickable build-libs --arch amd64``. Don't forget to mention this step in your README!
+
+See how `Teleports clickable.json <https://gitlab.com/ubports/apps/teleports/blob/master/clickable.json#L14>`_ uses the libraries feature to build its dependency tdlib.
+
+Pre-built
+^^^^^^^^^
+Note, that a pre-built library is usually only available as a shared object, that needs to be linked dynamically (no static linking). Furthermore, it may contain components that you don't need, resulting in a bloated app. It may even miss something, that you could achieve by compiling it yourself. Sometimes, a library is available in the Ubuntu Repositories, but is not installable for the architecture you need (likely ``armhf``). In this case you have to compile the library as described above.
```

Comment:
What does `(no static linking)` mean for me, a developer following this guide?

Also, the first comma is not needed.